pkg/controller/jobframework/reconciler.go

```diff
@@ -1020,3 +1022,14 @@ func clearMinCountsIfFeatureDisabled(in []kueue.PodSet) []kueue.PodSet {
 	}
 	return in
 }
+
+// FilterAnnotationsWithPrefix filters out annotations containing prefix.
```

Comment:
nit: `// filters for annotations containing prefix`

I think that out implies that you're removing annotations with the prefix